models/users.js

```diff
@@ -483,6 +483,28 @@ const getDiscordUsers = async () => {
   }
 };
 
+const updateRoles = async (userData, newRoles) => {
+  try {
+    const roles = { ...userData.roles };
+    const newRolesArray = Object.entries(newRoles);
+    const rolesContainNewRoles = newRolesArray.every(([key, value]) => {
+      return roles[key] === value;
+    });
```

Comment:
Tejas has asked that he will not merge any PRs with the `Generic Object Injection Sink` warning. can you please see if we can resolve this.